base/indices.jl

```diff
@@ -357,11 +357,16 @@ LinearIndices(A::Union{AbstractArray,SimpleVector}) = LinearIndices(axes(A))
 IndexStyle(::Type{<:LinearIndices}) = IndexLinear()
 axes(iter::LinearIndices) = iter.indices
 size(iter::LinearIndices) = map(unsafe_length, iter.indices)
-length(iter::LinearIndices) = prod(size(iter))
```

Comment:
If something is iterable and has a known length, it makes sense to define `length`.

Reply:
In this case it's also an `AbstractArray`, in which case exactly this definition is defined as a fallback.  I defined it because (like you) I was thinking about it as an iterable — but Milan is also correct that this is an array and doesn't technically need it.

Reply:
Ah of course; I see.